service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -315,6 +317,24 @@ protected final void addTlsTrustOptions(final NetServerOptions serverOptions) {
         }
     }
 
+    /**
+     * Configures the TLS engine for the given set of server options.
+     *
+     * @param serverOptions The options to add configuration to.
+     */
+    protected final void addTlsEngineOptions(final NetServerOptions serverOptions) {
+
+        if(getConfig().isNativeTlsEnabled()) {
```

Comment:
we should also check for `io.netty.handler.ssl.OpenSsl.isAvailable()` here to fall back to JDK ssl if OpenSSL cannot be found.

Reply:
I was asking myself the same question when implementing, but decided against the fallback. 
When configured with ``true`` a user would have to browse through the logs to find out that actually he isn't using the native implementation. Without the fallback there is an immediate feedback to the user. 

If you like the fallback more we, could then also think about enable the native implementation by default. 


WDYT?

Reply:
Yep, that's true. I renamed the property to {{nativeTlsRequired}}